src/services/chains.rs

```diff
@@ -0,0 +1,36 @@
+use crate::cache::cache_operations::RequestCached;
+use crate::config::{
+    base_config_service_url, chain_info_cache_duration, chain_info_request_timeout,
+};
+use crate::models::chains::ChainInfo;
+use crate::models::commons::Page;
+use crate::utils::context::Context;
+use crate::utils::errors::ApiResult;
+
+pub async fn get_chains_paginated(
+    context: &Context<'_>,
+    limit: Option<&String>,
+    offset: Option<&String>,
+) -> ApiResult<Page<ChainInfo>> {
+    let mut queries: Vec<(String, String)> = vec![];
+    if let Some(limit) = limit {
+        queries.push(("limit".to_string(), limit.to_owned()))
+    }
+    if let Some(offset) = offset {
+        queries.push(("offset".to_string(), offset.to_owned()))
+    }
+    let mut url = reqwest::Url::parse_with_params(base_config_service_url().as_str(), queries)
```

Comment:
we could import `Url` to avoid that we have to specify the `reqwest` module. (There are other places too where we should do that)